thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java

```diff
@@ -190,4 +165,88 @@ public Response computeSeverity(@NotNull @QueryParam("collection") String collec
 
     return Response.ok(severity.toString(), MediaType.TEXT_PLAIN_TYPE).build();
   }
+
+  /**
+   * Breaks down the given range into consecutive monitoring windows as per function definition
+   * Regenerates anomalies for each window separately
+   *
+   * As the anomaly result regeneration is a heavy job, we move the function from Dashboard to worker
+   * @param id anomaly function id
+   * @param startTimeIso The start time of the monitoring window (in ISO Format), ex: 2016-5-23T00:00:00Z
+   * @param endTimeIso The start time of the monitoring window (in ISO Format)
+   * @param isForceBackfill false to resume backfill from the latest left off
+   * @return HTTP response of this request
+   * @throws Exception
+   */
+  @POST
+  @Path("/{id}/generateAnomaliesInRange")
+  public Response generateAnomaliesInRange(@PathParam("id") String id,
+      @QueryParam("start") String startTimeIso,
+      @QueryParam("end") String endTimeIso,
+      @QueryParam("force") @DefaultValue("false") String isForceBackfill) throws Exception {
+    long functionId = Long.valueOf(id);
+    AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(functionId);
+    if (anomalyFunction == null) {
+      return Response.noContent().build();
+
+    }
+
+    // Check if the timestamps are available
+    DateTime startTime = null;
+    DateTime endTime = null;
+    if (startTimeIso == null || startTimeIso.isEmpty()) {
+      throw new IllegalArgumentException(String.format("[functionId %s] Monitoring start time is not found", id));
+    }
+    if (endTimeIso == null || endTimeIso.isEmpty()) {
+      throw new IllegalArgumentException(String.format("[functionId %s] Monitoring end time is not found", id));
+    }
+
+    startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(startTimeIso);
+    endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(endTimeIso);;
+
+    if(startTime.isAfter(endTime)){
+      throw new IllegalArgumentException(String.format(
+          "[functionId %s] Monitoring start time is after monitoring end time", id));
+    }
+    if(endTime.isAfterNow()){
```

Comment:
This is a very good checking! Could you improve it further and check the max date time of the dataset instead of the current time.